### PR TITLE
LPS-174744 Item selector URL should receive the current selected item

### PR DESCRIPTION
**Pull request by ambrinchaudhary**
@marcogalluzzi please let me know if this approach would work for you. 

This is working for the case described in https://issues.liferay.com/browse/LPS-153332 

This is the simplest way I have found to send to back the selected item. It does not imply to change any API (nor other teams to approve changes), but we should send this param whereever we want to open the folder where the last item was selected.

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

**Comment by liferay-lima** [comment]
ci:test:relevant

**Comment by ambrinchaudhary** [comment]
ci:stop

**Comment by ambrinchaudhary** [comment]
ci:test:sf

**Comment by marcogalluzzi** [comment]
Hi @ambrinchaudhary I could use the parameter to select the correct folder in the DL item selector, however:
* The portlet namespace next to the parameter name is not a good choice because in the DL item selector we don't know which portlet is going to call it. The used namespace in ImagePicker.es.js is `_com_liferay_journal_web_portlet_JournalPortlet_` but the current namespace in DL item selector is `_com_liferay_item_selector_web_portlet_ItemSelectorPortlet_`. I would replace it with the portlet namespace of the DL item selector or remove it completely.
* We still need to pass the computed folderId to the repository browser tag in order to show the correct breadcrumb

**Comment by ambrinchaudhary** [comment]
>  I would replace it with the portlet namespace of the DL item selector or remove it completely.
Then I think is better to remove it, I thought it was neccessary... 

> We still need to pass the computed folderId to the repository browser tag in order to show the correct breadcrumb
Can't you get that folderId in backend with the fileEntryId I pass to you?

**Comment by marcogalluzzi** [comment]
Hi @ambrinchaudhary now the breadcrumb shows the correct folder path. The final solution is more straightforward than adding the parameter "folderId" to a DynamicServletRequest.

However, IMHO it is important to select the correct tab of the selector, otherwise we can have a strange (navigable!) folder path like this (case: an image from Web Content Images has been selected, then reopen the Item Selector).

Captura de pantalla 2023-02-08 a las 12 54 12



Finally, before sending new changes, please remember to *pull* my changes.

**Comment by ambrinchaudhary** [comment]
> However, IMHO it is important to select the correct tab of the selector, otherwise we can have a strange (navigable!) folder path like this (case: an image from Web Content Images has been selected, then reopen the Item Selector).

I've created a separate subtask for this: https://issues.liferay.com/browse/LPS-175165

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/ImagePicker/ImagePicker.es.js [hunk 1]
```diff
@@ -99,7 +105,10 @@ const ImagePicker = ({
 				Liferay.Language.get('select-x'),
 				Liferay.Language.get('image')
 			),
-			url: itemSelectorURL,
+			url: addParams(
+				`selectedFileEntryId=${selectedImageId}`,
```
Comment (adolfopa) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/ImagePicker/ImagePicker.es.js]: Shouldn't we use a more generic name (e.g. `selectedEntryId`)? I think we'll need to apply similar changes to other item selectors that support folders (KB, Journal) where that param name doesn't make sense.

**Comment by adolfopa** [comment]
@ambrinchaudhary We'll be doing similar changes in other places right? This should work with any kind of document, not only images.

**Review by adolfopa — CHANGES_REQUESTED** [review]

**Comment by ambrinchaudhary** [comment]
> @ambrinchaudhary We'll be doing similar changes in other places right? This should work with any kind of document, not only images.

Yes, I've created another task for this https://issues.liferay.com/browse/LPS-175163. We would need to analyse and look for all the cases we want to support.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/ImagePicker/ImagePicker.es.js (hunk 1 above)
Comment (ambrinchaudhary) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/ImagePicker/ImagePicker.es.js]: I see.... what about `selectedItemId`? Anyway I still have doubts, what if we are opening a multiple selection Item Selector? Should we pass always an array to cover all the cases?

Comment (adolfopa) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/ImagePicker/ImagePicker.es.js]: `selectedItemIds` is fine too. Passing a single element is a special case of passing multiple ones, so its probably better to pass the array.

**Comment by ambrinchaudhary** [comment]
ci:test:sf

**Comment by ambrinchaudhary** [comment]
ci:test:relevant

**Comment by AliciaGarciaGarcia** [comment]
**Before** reviewing it.

Have to take account the repository?

_If you select the file from an asset lib, the selected item is from the repo and this kind of things_

Take this into account, because it will bite us on the... future...


Thanks!

**Comment by ambrinchaudhary** [comment]
> **Before** reviewing it.
> 
> Have to take account the repository?
> 
> _If you select the file from an asset lib, the selected item is from the repo and this kind of things_
> 
> Take this into account, because it will bite us on the... future... maybe add to this PR or create an task on the story/epic
> 
> Thanks!

Good point @AliciaGarciaGarcia . As this is under FF, we can create a separate sub-task to solve this. 
Would you still like to review this PR? Otherwise I will forward it. 
Thanks!

**Comment by AliciaGarciaGarcia** [comment]
hi @ambrinchaudhary 
I will review it, but I do not know if @marcogalluzzi is looking at it or not actually.

@marcogalluzzi this PR will be this way or are you looking at the repository right now?

thanks!

**Review by adolfopa — APPROVED** [review]

File: modules/apps/document-library/document-library-item-selector-web/src/main/java/com/liferay/document/library/item/selector/web/internal/display/context/DLItemSelectorViewDisplayContext.java [hunk 2]
```diff
@@ -368,16 +377,28 @@ private Long _getFileEntryTypeId() {
 			infoItemItemSelectorCriterion.getItemSubtype());
 	}
 
-	private long _getFolderId() {
-		if (_folderId != null) {
-			return _folderId;
+	private long _getFolderIdFromRequestParameters(
```
Comment (adolfopa) [inline modules/apps/document-library/document-library-item-selector-web/src/main/java/com/liferay/document/library/item/selector/web/internal/display/context/DLItemSelectorViewDisplayContext.java:380]: Just a minor thing: I'd prefer to keep the private method name simply as `_getFolderId`. Qualifying the name would make sense if there where different sources from where the folder ID was read, but it is always read from the request. I think it is redundant.

But I don't think this is a blocker, so I've approved the pr.

Comment (marcogalluzzi) [reply modules/apps/document-library/document-library-item-selector-web/src/main/java/com/liferay/document/library/item/selector/web/internal/display/context/DLItemSelectorViewDisplayContext.java]: Change done. I prefer to change the name now and not be asked later when merging to master 😅

**Comment by marcogalluzzi** [comment]
Hi @AliciaGarciaGarcia I've revised the current solution and as you already foreseen it doesn't work with an Asset Library root folder. We leave the solution for this problem to a new subtask: [LPS-175365](https://issues.liferay.com/browse/LPS-175365)

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Review by AliciaGarciaGarcia — APPROVED** [review]
working as expected on https://issues.liferay.com/browse/LPS-153332?focusedCommentId=2845702&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-2845702 with the FF activate

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Comment by AliciaGarciaGarcia** [comment]
ci:forward

**Pull request closed without merging**